src/car.hpp

```diff
@@ -45,13 +45,43 @@ class Car : public Renderable
 	// Set rotation in radians
 	void set_rotation(float radians);
 
+	// get or set the lane associated with car
+	void set_lane(int n);
+	int get_lane();
+
+	// car to slows down to stop sign or speeds away
+	void slow_down();
+	void speed_up();
+
+	// Set indicator for passage
+	void signal_to_move();
+
+	// get acceleration, velocity, max_speed
+	vec2 get_acc();
+	vec2 get_vel();
+	float get_max_speed();
+
+	// Intersection Detection
+	bool is_approaching_stop(vec2 lane_pos);
+	
+	// Calculation for safe stopping distance
+	float compute_stopping_dis(float velocity, float acc);
+
 private:
 	vec2 m_position; // Window coordinates
 	vec2 m_scale; // 1.f in each dimension. 1.f is as big as the associated texture
 	float m_rotation; // in radians
 	size_t m_num_indices; // passed to glDrawElements
-	float m_current_speed;
-	float m_x_dir;
-	float m_y_dir;
+	//float m_current_speed;
+	//float m_x_dir;
+	//float m_y_dir;

```

Comment:
should we just remove these? seems like you're handling all this stuff with different variables